Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/Stages.hpp

```diff
@@ -13,13 +13,20 @@
 #include <exception>
 #include <vector>
 
+#include <c10/cuda/CUDAStream.h>
 #include <torch/torch.h>
 
 namespace Acts {
 
 /// Error that is thrown if no edges are found
 struct NoEdgesError : std::exception {};
 
+/// Capture the context of the execution
+struct ExecutionContext {
+  torch::Device device{torch::kCPU};
+  std::optional<c10::cuda::CUDAStream> stream;
+};
```

Comment:
What is the meaning here if the device is set to CPU but the CUDA stream is set to something? Or is that an invalid state? What about if the device is set to a GPU but the stream is unset? Does it then use the default stream?

Reply:
Yeah so this definitively can be in an inivalid state. I wanted to keep it lightweight for now, but of course could add constructors to ensure validity of the state.
what happens if the stream is not set but we are on GPU is decided in the modules I would say.

Reply:
Okay!